src/main/scala/com/microsoft/hyperspace/Hyperspace.scala

```diff
@@ -74,7 +75,19 @@ class Hyperspace(spark: SparkSession) {
    * @param indexName Name of the index to refresh.
    */
   def refreshIndex(indexName: String): Unit = {
-    indexManager.refresh(indexName)
+    refreshIndex(indexName, REFRESH_MODE_FULL)
+  }
+
+  /**
+   * Update indexes for the latest version of the data. This api provides a few supported
+   * refresh modes as listed below.
+   *
+   * @param indexName Name of the index to refresh.
+   * @param mode Refresh mode. Currently supported modes are [[REFRESH_MODE_INCREMENTAL]] and
+   *             [[REFRESH_MODE_FULL]]
```

Comment:
Can we not use constants at this level similar to https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L997. Other languages will not be able to utilize it anyway, and we can just use them internally.

Reply:
thanks @imback82 , added